### ISSUE TITLE
Some DICOM volumes are loaded flipped along Z axis in ITK-v5.1rc01

### ISSUE THREAD

**Issue by lassoan** [InsightSoftwareConsortium/ITK#1587]
### Description

Some volumes are flipped along their Z axis (ordering of slices is inverted). Other axes are not inverted, so it is not a rotation but a mirroring along one axis. It worked well in the previous ITK 5 release.

![image](https://user-images.githubusercontent.com/307929/73612832-5d9e1c80-45bd-11ea-9fcf-996f745d2c9e.png)

### Steps to Reproduce

Load the DICOM volume using ITK-5.0 and ITK-5.1rc01 and compare them in a sagittal view.

### Expected behavior

They should look the same.

### Actual behavior

There is a flip along the IS axis.

### Reproducibility

I can provide an anonymized data set on request but it is not for public sharing.

### Versions

ITK-v5.1rc01

### Environment

Windows, Visual Studio 2015

**Comment by pieper** [comment]
I can reproduce with public data (this was series 7 from patient 0000 in [this TCIA collection](https://wiki.cancerimagingarchive.net/pages/viewpage.action?pageId=61081171).

Also can confirm that this appears on a Mac build.  I'll try to come up with a simple way to replicate the issue.

**Comment by jhlegarreta** [comment]
May be related to #1163 ?

**Comment by dzenanz** [comment]
No, #1163 is about different naming conventions for orientations, not orientations being different.

**Comment by lassoan** [comment]
We have done some more testing and found that file names matter: by renaming files from 0001, 0002, 003, ... to 0400, 0399, 0398, ... we could make the image flip upside down.

@pieper did some more testing and it seems that the issue is related to how the DICOM reader is used in Slicer (we collect the DICOM files and pass the filename list to ITK). Something has changed in ITK that changed the behavior, for example, the filename list was previously sorted internally in ITK, but now it is considered to be already sorted?

**Comment by dzenanz** [comment]
@malaterre and @issakomi might want to pitch in.

**Comment by pieper** [comment]
Some more detail on what @lassoan reported, when users load data through Slicer's DICOM module, we sort the data according to geometry (and do some other consistency checks) and send the sorted data to itk for reading.

When users drop an arbitrary file on the application, we delegate down through some very old logic that was created in the early days of ITK and Slicer which tries to guess what the user might want to do with the data.  It is in [this old code](https://github.com/Slicer/Slicer/blob/master/Libs/vtkITK/vtkITKArchetypeImageSeriesReader.cxx) or something related that things no longer work.  Most of the original authors are long gone, and this old spaghetti code should go too, but it's odd that it worked for the last 14 years and now something broke.

**Comment by issakomi** [comment]
> We have done some more testing and found that file names matter: by renaming files from 0001, 0002, 003, ... to 0400, 0399, 0398, ... we could make the image flip upside down.

May be the change, s. link
https://sourceforge.net/p/gdcm/gdcm/ci/2b481c069485d364c1160ef6eb14fa9e52e48879/

I only guess, sorry, i don't use this code in my app and didn't try to reproduce the issue.

**Comment by issakomi** [comment]
@lassoan 
 to test, may be, s.
https://github.com/InsightSoftwareConsortium/ITK/blob/e138b60f4af6d1b9c835f975e6f97db692192bee/Modules/ThirdParty/GDCM/src/gdcm/Source/MediaStorageAndFileFormat/gdcmSerieHelper.cxx#L442

```
+/*
  else if ( ImageNumberOrdering(fileSet ) )
  {
  return ;
  }
+*/


```

**Comment by malaterre** [comment]
> @lassoan
> to test, may be, s.
> https://github.com/InsightSoftwareConsortium/ITK/blob/e138b60f4af6d1b9c835f975e6f97db692192bee/Modules/ThirdParty/GDCM/src/gdcm/Source/MediaStorageAndFileFormat/gdcmSerieHelper.cxx#L442
> 
> ```
> +/*
>   else if ( ImageNumberOrdering(fileSet ) )
>   {
>   return ;
>   }
> +*/
> ```

This could be the issue here. But I was asked recently to resurect this logic to make an ITK test pass, so I would not know which way to go then.

**Comment by issakomi** [comment]
@thewtex @dzenanz The issue with flip using "Load Data" dialog or drag-and-drop (not "Load DICOM Data"), mentioned in this [post above](https://github.com/InsightSoftwareConsortium/ITK/issues/1587#issuecomment-582642785) is discussed in this [topic](https://discourse.slicer.org/t/cranial-and-caudal-directions-are-opposite-when-loading-sequence-of-dicom-images). The issue is not related to ITK.

**Comment by issakomi** [comment]
Not sure here was ever an issue with data set without IPP/IOP. It was only my speculation because i  had checked difference between ITK versions and there was only this change related. May be the issue could be closed.
